### DISCUSSION TITLE
Small wishes

### DISCUSSION THREAD

**Discussion by grylem** [ololx/quick-symlink#21] [Announcements]
Привет,
будет здорово, если вы сможете добавить:
английскую локализацию
отображение иконок в контекстном меню
возможность отключения именно контекстного меню.
объединить контекстное меню для ссылок.
сделать иконки(finder toolbar) более подходящими для Big Sur.
С уважением.

**Comment by ololx** [comment]
Добрый день, 
Спасибо за обратную связь. Я обязательно все изучу и дам Вам развернутый ответ по Вашему предложению.
Но уже сейчас могу сказать следующее:
1 - Английская локализация уже существует, однако я для нее указан все наименования на русском языке - это баг, который Вы заметили и я его исправлю сегодня - завтра.
2 - По поводу остальных пунктов я отвечу Вам на выходных, но скорее всего я реализую все Ваши предложения, кроме "объединить контекстное меню для ссылок"; "объединить контекстное меню для ссылок" - кажется не очень удобным с точки зрения пользования (если имеется ввиду именно контекстное меню по правому клику мыши). Однако Вы можете переубедить меня, елси покажете мне дизайн такого меню (как Вы предлагаете).

**Comment by ololx** [comment]
Исправил баг с локализацией - пожалуйста проверьте версию 0.10.1. Если поменять в системных настройках язык с Русского на Английский - то пункты меню должны отображаться уже в Английской локализации.

**Comment by grylem** [comment]
> Вы можете переубедить меня, елси покажете мне дизайн такого меню (как Вы предлагаете).

я и на остальных пунктах не настаиваю - предпочитаю для этих целей ком. строку. Просто показалось что тогда можно будет сократить дублирующиеся - copy/paste link...

> Исправил баг с локализацией

вместо hardlink actions - alias actions, это вносит путанницу т.к. под alias в finder понимается другое.

Если возможно, то не помешало бы добавить авторизацию , с целью создания ссылок в папках требующих прав администратора.

**Comment by ololx** [comment]
Тогда все еще проще, сегодня переименую пункт меню - остальное сделаю уже позже, как написал ранее.
Еще раз спасибо за дельные предложения, особенно с авторизацией.

**Comment by grylem** [comment]
Если есть интерес в расширении функцинала, можно добавить - заменить ссылку с абсолютным путем на относительный, исправить неверную ссылку и/или проверить ее на соответствие, заменить symlink на hardlink и  наоборот, добавить unhardlink и  т.д.

**Comment by ololx** [comment]
```
Если есть интерес в расширении функцинала, можно добавить - заменить ссылку с абсолютным путем на относительный, исправить неверную ссылку и/или проверить ее на соответствие, заменить symlink на hardlink и наоборот, добавить unhardlink и т.д.
```
Изначально просто сделал это расширение для себя, но на данный момент я открыт для подобных предложений. Вот этот пункт `исправить неверную ссылку и/или проверить ее на соответствие` остался последним в последней issue #3; остальное, что Вы предложили достаточно интересное. Я просто добавлю все эти пункты в TODO проекта перед тем как закрыть этот ISSUE; Если у Вас есть предложения по реализации предложенных пунктов - был бы рад их обсудить и узнать, так мне было бы проще реализовать их удобными не только мне.

**Comment by grylem** [comment]
Может есть смысл сделать полноценную программу для работы с finder alias, symlinks, hardlinks в которой finder extensions будут только частью программы? Набор операций не такой уж большой и тогда контекстное меню не будет перегруженным.

**Comment by ololx** [comment]
Это у меня прописано в TODO, но пока руки не доходят) думаю сейчас уложу в голове Ваши предложения и попробую реализовать. Бонус от того, что эта штука нужна в основном мне) Пока я пытаюсь удобно объединить менюшки для ссылок, но все равно же выходит некоторое дублирование:
'''
Копировать путь отсюда
Вставить символьную ссылку сюда
Вставить жесткую ссылку сюда
Создать символьную ссылку
Создать жесткую ссылку
...
'''

**Comment by grylem** [comment]
> Это у меня прописано в TODO

извиняюсь, не прочитал, исправлю.

> Бонус от того, что эта штука нужна в основном мне

судя по issues не только вам. 

Посмотрите консольные программы symlinks, unhardlink. Они практически полностью закрывают потребности по операциям с symlinks и hardlinks.  Но у GUI (то что Вы сейчас) делаете поклонников больше чем у ком. строки.

**Comment by ololx** [comment]
Благодарю за совет. После завтра уже смогу нормально подумать над всеми предложениями и планом реализации - надо немного уложить в голове.

**Comment by grylem** [comment]
из TODO
> a2) create new hard link for selected folders and files

про особенности удаления жестких ссылок для папок на HFS вы знаете?

> a3) create new hard link alias for selected folders and files

речь о finder alias или о hard links?

**Comment by ololx** [comment]
> a2) create new hard link for selected folders and files
> про особенности удаления жестких ссылок для папок на HFS вы знаете?

Да.

> a3) create new hard link alias for selected folders and files
> речь о finder alias или о hard links? 
Речь о finder alias

Вроде бы учел все из всех ISSUE. Вероятно эту можно уже закрывать
Еще раз спасибо за идеи для развития функционала

Я еще хотел бы попросить Вас добавить себя в список контрибьютеров за помощь в развитие проекта, если Вы конечно не против.
